lib/Listener/RegisterOperationsListener.php

```diff
@@ -31,21 +31,29 @@
 use OCP\EventDispatcher\IEventListener;
 use OCP\Util;
 use OCP\WorkflowEngine\Events\RegisterOperationsEvent;
+use OCP\App\IAppManager;
+use Psr\Log\LoggerInterface;
 
 /**
  * @template-implements IEventListener<Event|RegisterOperationsEvent>
  */
 class RegisterOperationsListener implements IEventListener {
 	public function __construct(
 		protected readonly Operation $operation,
+		protected IAppManager $appManager
 	) {
 	}
 
 	public function handle(Event $event): void {
 		if (!$event instanceof RegisterOperationsEvent) {
 			return;
+		} else if(!$this->appManager->isInstalled('notifications') || 
+		!$this->appManager->isEnabledForUser('notifications')) {
+			$logger = \OC::$server->get(LoggerInterface::class);
```

Comment:
please either inject the LoggerInterface in the constructor, or use `\OCP\Log\logger(Application::APP_ID)->error(…)`

Reply:
Done!

Reply:
Thanks! I pushed a followup commit for the style, that is not reflected in the cs-fixer rules yet, to have the operators on a new line.

Reply:
… on hindsight, the isInstalled check is not necessary, having `!$this->appManager->isEnabledForUser('notifications')` is already sufficient. 

Also, a brief hint into the descriptions https://github.com/nextcloud/flow_notifications/blob/master/appinfo/info.xml#L11 that notifications app is required could be a good addition, perhaps also into the Readme.

Reply:
Good point. You can't get isEnabledForUser true if the app wasn't installed. Code updated!